public/app/features/dashboard-scene/utils/dashboardSceneGraph.ts

```diff
@@ -128,6 +136,14 @@ export function getNextPanelId(dashboard: DashboardScene): number {
   return max + 1;
 }
 
+export function getFilterAndGroupByVariables(scene: DashboardScene): Array<AdHocFiltersVariable | GroupByVariable> {
+  const vars = scene.state.$variables?.state.variables.filter(
+    (v) => v instanceof AdHocFiltersVariable || v instanceof GroupByVariable
+  ) as Array<AdHocFiltersVariable | GroupByVariable>;
```

Comment:
```suggestion
  const vars = scene.state.$variables?.state.variables.filter(
    (v): v is AdHocFiltersVariable | GroupByVariable => v instanceof AdHocFiltersVariable || v instanceof GroupByVariable
  );
```

think this is slightly better. still not great, but hopefully a future version of typescript will preserve these types properly 🤞